scripts/set_config.py

```diff
@@ -105,14 +104,19 @@ def parse_args(argv):
         )
     return parser.parse_args(argv)
 
-def create_config_obj(argv, log=None):
+def create_config_file(argv, config_dict=None, log_name=None):
```

Comment:
When we call this function from a python script, its name is a bit misleading. It is creating a file, not an object. It's also nice to be able to pass a dict when we don't have a file written out to disk.

I plan to open an issue about how this should be refactored for easier use as a function outside the CLI script.

Reply:
It sounds like this function is not even necessarily creating a config file. Is it possible that it might do nothing at all, e.g. if the configuration doesn't call for an output file and no `config_dict` is provided? If so, then I think this is still misnamed.

The length of this function suggests to me that it's trying to do too much, which makes naming it hard (not to mention testing it, or writing more than a not-very-informative one-liner docstring). I think that would all improve were it broken up. That's probably not something for this PR, but I think we should consider it for future work. I'd be happy to contribute.

Reply:
Just to be clear, @maddenp-noaa are you okay with me leaving the `create_config_file`  name here?

When it's incorporated into the SRW workflow, creating a config file is exclusively what it is supposed to do. The additional options and perks from the CLI are primarily for added user benefit outside the workflow.

Reply:
@christinaholtNOAA My complaint is that, unless I'm mistaken, this function can be called in such a way that it does nothing at all. That shouldn't be possible. It should either create a config file, or populate an in-memory config object. I think the issue stems from that fact that this function is doing too much. Better, IMO, would be to break this up into _at least_ two functions -- one that constructs an in-memory config object and returns it (a cleaner pattern than taking un- or semi-initialized objects as arguments and then mutating, a Python antipattern I believe appears elsewhere in this codebase), and another that takes such an object and writes it to a file. Callers can then call the function with the desired behavior, rather than implying it via a combination of set/unset function arguments. A third function could combine the two behaviors -- call the cfg object creator, then pass the cfg object to the writer -- as a convenience. I think there may be a couple more functions that could usefully be factored out of this big one, too.

I won't hold up this PR for this, but I think we should address this, as this is a crucial bit of logic, and currently too complex due to bypassing single-responsibility principle. It should be possible to simplify the tests, too, by refactoring this.